airbrakes/data_handling/data_processor.py

```diff
@@ -190,15 +190,16 @@ def _calculate_speed(self, a_x: list[float], a_y: list[float], a_z: list[float])
         Calculates the speed of the rocket based on the linear acceleration.
         Integrates the linear acceleration to get the speed.
         """
+        previous_vel_x, previous_vel_y, previous_vel_z = self._previous_velocity
         # We need at least two data points to calculate the speed:
         if len(self._data_points) < 2:
-            return 0.0
+            return np.sqrt(previous_vel_x**2 + previous_vel_y**2 + previous_vel_z**2)
 
         # Side note: We can't use the pressure altitude to calculate the speed, since the pressure
         # does not update fast enough to give us a good estimate of the speed.
 
         # calculate the time differences between each data point
-        time_diff = np.diff([data_point.timestamp for data_point in self._data_points])
+        time_diff = np.diff([data_point.timestamp for data_point in self._data_points]) * 10e-9
```

Comment:
should probably comment we are converting from ns to s

Reply:
fine